catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/NsiliEndpoint.java

```diff
@@ -167,22 +158,73 @@ public boolean getEnterpriseSearch() {
 
     public void setFilterBuilder(FilterBuilder filterBuilder) {
         this.filterBuilder = filterBuilder;
+        if (library != null) {
+            library.setFilterBuilder(filterBuilder);
+        }
     }
 
     public void setMaxPendingResults(int maxPendingResults) {
         this.maxPendingResults = maxPendingResults;
+        if (library != null) {
+            library.setMaxPendingResults(maxPendingResults);
+        }
     }
 
-    public void init() {
+    public void setCorbaOrb(CorbaOrb corbaOrb) {
+        this.corbaORB = corbaOrb;
+        this.orb = corbaOrb.getOrb();
+        corbaOrb.addCorbaServiceListener(this);
+    }
+
+    public void setOrb(ORB orb) {
+        this.orb = orb;
+    }
+
+    public void destroy() {
+        LOGGER.debug("Destroying NSILI Endpoint");
+        if (corbaORB != null) {
+            corbaORB.removeCorbaServiceListener(this);
+        }
+        library = null;
+        iorString = "";
+    }
+
+    @Override
+    public void corbaInitialized() {
         try {
-            if (orb == null) {
-                orb = getOrbForServer(corbaPort);
-                orbRunThread = new Thread(() -> orb.run());
-                orbRunThread.start();
-                LOGGER.info("Started ORB on port: {}", corbaPort);
+            orb = corbaORB.getOrb();
+            initCorba();
+        } catch (InvalidName | AdapterInactive | WrongPolicy | ServantNotActive | IOException | SecurityServiceException e) {
+            LOGGER.error("Unable to initialize Corba connection {}", e);
+        }
+    }
+
+    @Override
+    public void corbaShutdown() {
+        shutdownCorbaOrb();
```

Comment:
can this just have the logic of shutdownCorbaOrb instead of two functions?  seems redundant